### ISSUE TITLE
Incorrect "cannot access member x for type y" error

### ISSUE THREAD

**Issue by moshiur-raj** [microsoft/pyright#3294]
**Description**
pyright incorrectly shows this error. The code runs fine and plot() is a valid member of ax as stated in the [doc](https://matplotlib.org/3.5.0/api/figure_api.html#matplotlib.figure.Figure.add_subplot)
![image](https://user-images.githubusercontent.com/36466700/161399914-1fe5aeec-2f02-4c44-9e54-532d329cef9c.png)

**Code**
```python
import matplotlib.pyplot as plt

fig = plt.figure()
ax = fig.add_subplot(1, 1, 1)
ax.plot([0, 1], [0, 1])
plt.show()
```

**Additional context**
pyright version 1.1.234
os - archlinux

**Comment by erictraut** [comment]
Are you using any type stubs for matplotlib? I don't think it's a typed library. Which version of the library are you using? Have you enabled the `useLibraryCodeForTypes` configuration option?

**Comment by erictraut** [comment]
I'm able to repro what you're seeing if I enable `useLibraryCodeForTypes` and do not install any type stubs for matplotlib. We don't recommend enabling `useLibraryCodeForTypes` for static type checking because it tells pyright to infer types from an untyped (unannotated) library, but inferred types are often inaccurate or incomplete, especially in cases where the function is polymorphic (i.e. its return type depends on the input parameters).

Your options:
1. Switch from pyright to pylance, which includes minimal type stubs for matplotlib.
2. Find and install type stubs for matplotlib. For example the stubs that ship with pylance can be found [here](https://github.com/microsoft/python-type-stubs/tree/main/matplotlib).
3. Create your own type stubs for matplotlib that provide annotated methods and functions for the functionality you use.
4. Disable `usLibraryCodeForTypes`.
5. Ignore the error.

**Comment by moshiur-raj** [comment]
Thank you. I installed the type stubs from pylance.
